### PR TITLE
Nanodrop data source ids

### PR DESCRIPTION
**Pull request by bwolfe-benchling**
Fix problem where data source IDs were being used that did not refer to any existing measurement ID.

### CODE REVIEW

File: tests/parsers/thermo_fisher_nanodrop_eight/testdata/Thermo_NanoDrop_Eight_example01.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
                     "measurement time": "2020-05-26T15:36:00+00:00",
                     "measurement document": [
                         {
-                            "measurement identifier": "1fc444d9-389e-4df9-8a82-54460605f853",
+                            "measurement identifier": "f692658e-62f2-4383-8770-f0d97d2c17df",
```
Comment (bdworth) [inline tests/parsers/thermo_fisher_nanodrop_eight/testdata/Thermo_NanoDrop_Eight_example01.json:10]: Hopefully #296 will merge before this, in which case none of these ids will change anymore!

File: src/allotropy/parsers/thermo_fisher_nanodrop_eight/nanodrop_eight_parser.py [hunk 2]
```diff
@@ -163,41 +163,62 @@ def _get_calculated_data_document(
...
+        ):
+            data_source_doc_items.append(
+                DataSourceDocumentItem(
+                    data_source_feature="absorbance",
+                    data_source_identifier=_get_str(data, row, "a280 uuid"),
+                )
+            )
+
+        data_source_doc = None
+        if len(data_source_doc_items):
```
Comment (bdworth) [inline src/allotropy/parsers/thermo_fisher_nanodrop_eight/nanodrop_eight_parser.py:186]:
```suggestion
        if data_source_doc_items:
```

File: src/allotropy/parsers/thermo_fisher_nanodrop_eight/nanodrop_eight_parser.py [hunk 3]
```diff
@@ -163,41 +163,62 @@ def _get_calculated_data_document(
...
+            data_source_doc = DataSourceAggregateDocument(
+                data_source_document=data_source_doc_items
```
Comment (bdworth) [inline src/allotropy/parsers/thermo_fisher_nanodrop_eight/nanodrop_eight_parser.py:188]: Could you rename `data_source_doc` to `data_source_aggregated_document`? It's a bit confusing to read this as is.

File: src/allotropy/parsers/thermo_fisher_nanodrop_eight/nanodrop_eight_parser.py [hunk 4]
```diff
@@ -163,41 +163,62 @@ def _get_calculated_data_document(
...
+                DataSourceDocumentItem(
+                    data_source_feature="absorbance",
+                    data_source_identifier=_get_str(data, row, "a260 uuid"),
+                )
+            )
+
+        data_source_doc = None
+        if len(data_source_doc_items):
+            data_source_doc = DataSourceAggregateDocument(
+                data_source_document=data_source_doc_items
```
Comment (bdworth) [inline src/allotropy/parsers/thermo_fisher_nanodrop_eight/nanodrop_eight_parser.py:213]: same comments as above

File: tests/parsers/thermo_fisher_nanodrop_eight/thermo_fisher_nanodrop_eight_parser_test.py [hunk 5]
```diff
@@ -26,3 +26,30 @@ def test_parse_thermo_fisher_nanodrop_eight_to_asm_expected_contents(
...
+        for spec_doc in allotrope_dict["spectrophotometry aggregate document"][
+            "spectrophotometry document"
+        ]
+        for meas_doc in spec_doc["measurement aggregate document"][
+            "measurement document"
+        ]
+    ]
+
+    for data_source_id in data_source_ids:
+        assert data_source_id in measurement_ids
```
Comment (bdworth) [inline tests/parsers/thermo_fisher_nanodrop_eight/thermo_fisher_nanodrop_eight_parser_test.py:55]: If this test fails, we won't get a very helpful error message. Could you add one?

**Review by bdworth — CHANGES_REQUESTED** [review]
mostly lgtm, but a few small comments

**Review by nathan-stender — APPROVED** [review]

**Pull request merged**